File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.py
class Solution:
    def longestSubsequence(self, nums):
        n = len(nums)

        xor = 0
        has_non_zero = False

        for x in nums:
            xor ^= x

            if x != 0:
                has_non_zero = True

        # Entire array has non-zero XOR
        if xor != 0:
            return n

        # All elements are zero
        if not has_non_zero:
            return 0

        # Remove one non-zero element
        return n - 1

File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.js
var longestSubsequence = function(nums) {

    const n = nums.length;

    let xor = 0;
    let hasNonZero = false;

    for (const x of nums) {
        xor ^= x;

        if (x !== 0) {
            hasNonZero = true;
        }
    }

    // Entire array has non-zero XOR
    if (xor !== 0) {
        return n;
    }

    // All elements are zero
    if (!hasNonZero) {
        return 0;
    }

    // Remove one non-zero element
    return n - 1;
};